### DISCUSSION TITLE
Blocky on OPNSenese

### DISCUSSION THREAD

**Discussion by mihakralj** [mimugmail/opn-repo#35] [Ideas]
Blocky is a relatively new effort to make a modern pihole alternative. I compiled it on my OPNsense today (btw, do we have golang in ports yet? Blocky is written in go...) and it functions well. In the current iteration it requires Grafana to visualize DNS requests.

No rush, but we should keep an eye on Blocky and eventually add it into the repo.
https://github.com/0xERR0R/blocky/

**Comment by mihakralj** [comment]
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=255243

**Comment by mimugmail** [comment]
Fancy, when the port is in I will add it to the repo

**Comment by mihakralj** [comment]
Blocky is now available in FreeBSD repo - according to gregf here: https://github.com/0xERR0R/blocky/discussions/187
Can we add it to our repo, I will then write a UI wrapper and a template to generate config?

**Comment by mimugmail** [comment]
When Franco syncs the Ports I will build a pkg. Currently on mobile, need to check if its already in OPN ports

**Reply by mihakralj** [reply]
no rush, take your time - I need to build the GUI and that will take some time

**Reply by mimugmail** [reply]
Blocky is in the repo now:

```
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
Updating mimugmail repository catalogue...
Fetching meta.conf: 100%    163 B   0.2kB/s    00:01
mimugmail repository is up to date.
All repositories are up to date.
The following 1 package(s) will be affected (of 0 checked):

New packages to be INSTALLED:
        blocky: 0.14 [mimugmail]

Number of packages to be installed: 1

The process will require 13 MiB more space.
4 MiB to be downloaded.

Proceed with this action? [y/N]: y
[1/1] Fetching blocky-0.14.txz: 100%    4 MiB   3.7MB/s    00:01
Checking integrity... done (0 conflicting)
[1/1] Installing blocky-0.14...
[1/1] Extracting blocky-0.14: 100%
=====
Message from blocky-0.14:

--
A sample configuration file is installed at the following location:
/usr/local/etc/blocky-config.yml.sample

Default location for configuration file when using rc.d script:
/usr/local/etc/blocky-config.yml

Please refer to the documentation located at
https://0xerr0r.github.io/blocky/ for further information.

```

**Reply by justfly1111** [reply]
I  cant access it in the repo

**Reply by mimugmail** [reply]
I cant help without details

**Reply by maltechx** [reply]
I can't find the package either. When I list all the packages in the mimugmail repo, I get many results, but no blocky package. I'm running the latest 22.1.3 release. Maybe I'm missing something.
```
root@router:~ # cat /usr/local/etc/pkg/repos/mimugmail.conf
mimugmail: {
  url: "https://opn-repo.routerperformance.net/repo/${ABI}",
  priority: 190,
  enabled: yes
}

root@router:~ # pkg update
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
Updating mimugmail repository catalogue...
mimugmail repository is up to date.
All repositories are up to date.
root@router:~ # pkg search blocky
root@router:~ # 
```

**Reply by das-s** [reply]
I'm also unable to find the package running a very similar setup to maltechx.

**Reply by aism** [reply]
Same for me as well

**Comment by gspannu** [comment]
Is Blocky available on OPNsense yet?
Any **instructions as to how to install and run** as an alternative to Unbound?

Thanks…

**Comment by gspannu** [comment]
@mimugmail / @mihakralj 
Is the Blocky DNS available as part of mimi gmail repo? 
Or any **instructions as how to how I can install it within OPNsense**?

Many thanks…

**Comment by asche77** [comment]
A Blocky plugin for opnsense is provided here:

https://forum.opnsense.org/index.php?topic=42631.0

**Reply by brokoler** [reply]
> A Blocky plugin for opnsense is provided here:
> 
> https://forum.opnsense.org/index.php?topic=42631.0

Yes, this plugin is from the maintainer "gspannu" but it's not updated frequently.
Based on that a more up to date solution would be preferred, really hope it's someday added to opn-repo.

**Reply by gspannu** [reply]
**I have updated the Blocky plugin** to reflect the **latest Blocky 0.33.0**...
Will update the rest once 26.7 gets released shortly...
